rbac/management/group/relation_api_dual_write_group_handler.py

```diff
@@ -141,9 +141,15 @@ def replicate_added_role(self, role: Role):
             # There is a risk of phantom reads here, but this is why we lock the custom Role
             bindings: Iterable[BindingMapping] = BindingMapping.objects.filter(role=role).select_for_update()
 
-            for mapping in bindings:
-                self.group_relations_to_add.append(mapping.add_group_to_bindings(str(self.group.uuid)))
-                mapping.save(force_update=True)
+            if sum(1 for _ in bindings) == 0:
+                mapping = BindingMapping.for_group_and_role(role, str(self.group.uuid), self.group.tenant.org_id)
```

Comment:
Should we create binding mappings for the role if none exist, similar to how it is done for system roles?

Reply:
No, because binding mappings are per role and resource, the mappings would already exist for the role. This would not calculate/persist anything that would and should have already been calculated or persisted. If there are no existing role bindings, it means there is no access defined for the role. When access is later defined for the role, policies will be looked up then and the groups should be added. Would be good to add a test case for this in the test_dual_write test cases.